### PR TITLE
Fix docstring for std.fs.path.extension

### PR DESCRIPTION
**Pull request by yujiri8**
Seems like an accidentally unfinished sentence in here. I shortened it and added an example.

### CODE REVIEW

File: lib/std/fs/path.zig [hunk 1]
```diff
@@ -1271,11 +1271,12 @@ fn testRelativeWindows(from: []const u8, to: []const u8, expected_output: []cons
 
 /// Returns the extension of the file name (if any).
 /// This function will search for the file extension (separated by a `.`) and will return the text after the `.`.
-/// Files that end with `.` are considered to have no extension, files that start with `.`
+/// Files that start or end with `.` are considered to have no extension.
```
Comment (squeek502) [inline lib/std/fs/path.zig:1274]: How about:

```
/// Files that end with `.` are considered to have no extension.
/// Files that start with `.` and have no other `.` in their file name are considered to have no extension.
```

Seems more precise/accurate considering the `.gitignore => ""`, `.image.png => ".png"` examples.

Comment (yujiri8) [reply lib/std/fs/path.zig]: Good idea. I prefer one sentence though. I changed the docstring and added the test case.

File: lib/std/fs/path.zig [hunk 2]
```diff
@@ -1271,11 +1271,13 @@ fn testRelativeWindows(from: []const u8, to: []const u8, expected_output: []cons
 
 /// Returns the extension of the file name (if any).
 /// This function will search for the file extension (separated by a `.`) and will return the text after the `.`.
-/// Files that end with `.` are considered to have no extension, files that start with `.`
+/// Files that start with `.`, or that end with `.` and have no other `.` in their name,
```
Comment (squeek502) [inline lib/std/fs/path.zig:1274]: This is backwards.

```suggestion
/// Files that start with `.`, or that end with `.` and have no other `.` in their name,
/// Files that end with `.`, or that start with `.` and have no other `.` in their name,
```

Comment (yujiri8) [reply lib/std/fs/path.zig]: Dammit. Thank you, updated again.

**Pull request merged**